Libraries/LibJS/Parser.h

```diff
@@ -83,9 +84,19 @@ class Parser {
     void expected(const char* what);
     Token consume();
     Token consume(TokenType type);
+    void save_state();
```

Comment:
@awesomekling What do you think of this API? I found something like this necessary because it's difficult to tell the difference an arrow function argument list and an expression in parentheses without some way to rollback the parser.

Reply:
Seems okay. :)